qiskit_ibm_runtime/utils/validations.py

```diff
@@ -126,3 +130,39 @@ def validate_job_tags(job_tags: Optional[List[str]]) -> None:
         not isinstance(job_tags, list) or not all(isinstance(tag, str) for tag in job_tags)
     ):
         raise IBMInputValueError("job_tags needs to be a list of strings.")
+
+
+def validate_exec_spans_in_result(result: PrimitiveResult) -> bool:
+    """Validate execution span section in result metadata.
+
+    Args:
+        result: A primitive result to be validated
+
+    Returns True if validation succeeds
+    """
+
+    if (
+        "execution" not in result.metadata
+        or not isinstance(result.metadata["execution"], dict)
+        or "execution_spans" not in result.metadata["execution"]
+        or not isinstance(result.metadata["execution"]["execution_spans"], ExecutionSpanSet)
+    ):
+        return False
+
+    slice_ends = [0] * len(result)
+    for exspan in result.metadata["execution"]["execution_spans"]:
+        for task_id, task_slice in exspan.data_slices.items():
+            if task_slice.start != slice_ends[task_id]:
+                return False
+            slice_ends[task_id] = task_slice.stop
+
+    for pub_length, res in zip(slice_ends, result):
+        if len(res.data.shape) == 0:
+            expected_length = 1
+        else:
+            expected_length = prod(res.data.shape)
```

Comment:
math.prod returns 1 for zero-length inputs
```suggestion
        expected_length = prod(res.data.shape)
```